### ISSUE TITLE
[Bug]: Speculative decoding with a draft model makes generation slower

### ISSUE THREAD

**Issue by maiiabocharova** [vllm-project/vllm#15025]
### Your current environment

I tried several vLLM versions (0.6.2 and latest 0.7.3) and have consistent speed drop when using speculative decoding with draft model

### 🐛 Describe the bug

Main model - `1.7B,` speculative model - `135M` parameters (`SmolLMv2` family of models)
I trained the small model using logits distillation of main model, so it has a good level of generation (acceptance rate is very high)

Still I get consistent performance drop ~30% in terms of speed when using 5 speculative tokens, when I reduce number speculative tokens - speed increases, but the best speed in achieved when using main model only without speculative.

Here are my parameters:
```python
llm = LLM(
    model=MODEL_PATH,
    speculative_model=SPECULATIVE_MODEL_PATH,
    max_model_len=2500,
    num_speculative_tokens=5,
    gpu_memory_utilization=0.9
)

sampling_params = SamplingParams(
            temperature=0,
            top_k=1,
            max_tokens=256,
        )

outputs = llm.generate(prompts, sampling_params)
```

Is it expected?

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by hongjx175** [comment]
same problem

**Comment by CubieX** [comment]
Same. Using DeepSeek 32B Distill Quen + 1.5B DeekSeek Distill Quen for Draft.
Got about 30-50% less speed of the non-speculative decoding tests.
Using 0 (min) 16 (max) draft size. With 0.75 cutoff.
